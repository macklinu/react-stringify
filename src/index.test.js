// @flow

import React from 'react'
import { create } from 'react-test-renderer'
import Stringify from '.'
import type { Props } from '.'

const exampleValue = {
  a: 1,
  b: 2,
  nested: {
    c: 3,
    d: 4,
  },
}

test('returns null if undefined value passed in', () => {
  expect(renderStringify()).toBeNull()
})
test('honors falsy values besides undefined', () => {
  expect(renderStringify({ value: null })).toMatchSnapshot()
  expect(renderStringify({ value: 0 })).toMatchSnapshot()
  expect(renderStringify({ value: false })).toMatchSnapshot()
  expect(renderStringify({ value: '' })).toMatchSnapshot()
})

test('renders array values', () => {
  expect(renderStringify({ value: [1, 2, 3] })).toMatchSnapshot()
})

test('ignores functions in object by default', () => {
  expect(
    renderStringify({
      value: {
        foo: () => 'foo',
        bar: 'bar',
      },
    })
  ).toMatchSnapshot()
})

test('uses default render function if none passed', () => {
  expect(renderStringify({ value: exampleValue })).toMatchSnapshot()
})

test('honors space prop', () => {
  expect(
    renderStringify({ value: exampleValue, space: '\t' })
  ).toMatchSnapshot()
})

test('honors replacer prop', () => {
  expect(
    renderStringify({
      value: exampleValue,
      replacer(key, value) {
        if (key === 'nested') {
          return
        }
        return value
      },
    })
  ).toMatchSnapshot()
})

test('honors render prop API', () => {
  expect(
    renderStringify({
      value: exampleValue,
      children(string) {
        return <div>{string}</div>
      },
    })
  ).toMatchSnapshot()
})

function renderStringify(props?: Props = { value: undefined }) {
  return create(<Stringify {...props} />).toJSON()
}
