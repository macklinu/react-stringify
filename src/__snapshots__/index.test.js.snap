// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`honors falsy values besides undefined 1`] = `
<pre>
  null
</pre>
`;

exports[`honors falsy values besides undefined 2`] = `
<pre>
  0
</pre>
`;

exports[`honors falsy values besides undefined 3`] = `
<pre>
  false
</pre>
`;

exports[`honors falsy values besides undefined 4`] = `
<pre>
  ""
</pre>
`;

exports[`honors render prop API 1`] = `
<div>
  {
  "a": 1,
  "b": 2,
  "nested": {
    "c": 3,
    "d": 4
  }
}
</div>
`;

exports[`honors replacer prop 1`] = `
<pre>
  {
  "a": 1,
  "b": 2
}
</pre>
`;

exports[`honors space prop 1`] = `
<pre>
  {
	"a": 1,
	"b": 2,
	"nested": {
		"c": 3,
		"d": 4
	}
}
</pre>
`;

exports[`ignores functions in object by default 1`] = `
<pre>
  {
  "bar": "bar"
}
</pre>
`;

exports[`renders array values 1`] = `
<pre>
  [
  1,
  2,
  3
]
</pre>
`;

exports[`uses default render function if none passed 1`] = `
<pre>
  {
  "a": 1,
  "b": 2,
  "nested": {
    "c": 3,
    "d": 4
  }
}
</pre>
`;
